server/app/views/components/LinkElement.java

```diff
@@ -106,79 +112,98 @@ public LinkElement opensInNewTab() {
     return this;
   }
 
-  public ContainerTag asAnchorText() {
-    ContainerTag tag = Strings.isNullOrEmpty(href) ? div(text) : a(text).withHref(href);
+  public ATag asAnchorText() {
+    ATag tag = a(text);
     return tag.withCondId(!Strings.isNullOrEmpty(id), id)
         .withCondHref(!Strings.isNullOrEmpty(href), href)
         .withCondTarget(doesOpenInNewTab, "_blank")
         .withClasses(DEFAULT_LINK_STYLES, styles);
   }
 
-  public ContainerTag asButton() {
-    ContainerTag tag =
-        Strings.isNullOrEmpty(href)
-            ? div(text)
-            : a(text).withHref(href).withCondTarget(doesOpenInNewTab, "_blank");
-    return tag.withCondId(!Strings.isNullOrEmpty(id), id)
-        .withClasses(DEFAULT_LINK_BUTTON_STYLES, styles);
+  private <T extends ContainerTag<T> & IHref<T> & ITarget<T>> void setTargetMaybeHref(T tag)
+      throws RuntimeException {
```

Comment:
Not blocking: There's no need to explicitly declare this as it's an unchecked exeception.

Reply:
so you mean get rid of the `throws RuntimeException`?

Reply:
Yes, sorry about that.